### PR TITLE
Changed field definition to a pre-solver step

### PR DESCRIPTION
**Pull request by mmcgurn**
This PR changed the field specification to a pre solver step allowing us to use the same fields in different solvers.

### CODE REVIEW

File: ablateLibrary/domain/fieldDescription.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+PetscObject ablate::domain::FieldDescription::CreatePetscField(DM dm) const {
+    switch (type) {
+        case FieldType::FEM: {
+            // determine if it a simplex element and the number of dimensions
+            DMPolytopeType ct;
+            PetscInt cStart;
+            DMPlexGetHeightStratum(dm, 0, &cStart, NULL) >> checkError;
+            DMPlexGetCellType(dm, cStart, &ct) >> checkError;
+            PetscInt simplex = DMPolytopeTypeGetNumVertices(ct) == DMPolytopeTypeGetDim(ct) + 1 ? PETSC_TRUE : PETSC_FALSE;
```
Comment (knepley) [inline ablateLibrary/domain/fieldDescription.cpp:30]: You can just use DMPlexIsSimplex() here I think.

File: ablateLibrary/domain/fileMesh.cpp [hunk 2]
```diff
@@ -1,30 +1,24 @@
 #include "fileMesh.hpp"
 #include <utilities/petscError.hpp>
 #include <utilities/petscOptions.hpp>
-ablate::domain::FileMesh::FileMesh(std::string nameIn, std::filesystem::path pathIn, std::shared_ptr<parameters::Parameters> options, std::vector<std::shared_ptr<modifier::Modifier>> modifiers)
```
Comment (knepley) [inline ablateLibrary/domain/fileMesh.cpp:4]: Note that you can do this using `-dm_plex_filename`, so it is unclear whether we need this class, but maybe due to how the YAML is processed.

**Review by knepley — APPROVED** [review]
I tried to go over everything. I think it looks good. I made a few comments.

File: ablateLibrary/domain/fileMesh.cpp (hunk 2 above)
Comment (mmcgurn) [reply ablateLibrary/domain/fileMesh.cpp:4]: Thanks!  It is not really needed, it must simplifies the setup in yaml so yaml is self explaining for the examples.

File: ablateLibrary/domain/fieldDescription.cpp (hunk 1 above)
Comment (mmcgurn) [reply ablateLibrary/domain/fieldDescription.cpp]: Thanks!  I updated the code.

**Pull request merged**